### PR TITLE
fix(client): improve output for incompatible system object

### PR DESCRIPTION
**Pull request by mlegner**

### CODE REVIEW

File: Cargo.lock
Comment (giac-mysten) [inline Cargo.lock:1]: Question: many changes to the cargo lock. Are these expected?

File: crates/walrus-sui/src/client/read_client.rs [hunk 1]
```diff
@@ -297,13 +297,22 @@ async fn get_system_package_id(
...
-        .map_err(|_| SuiClientError::WalrusSystemObjectDoesNotExist(system_object_id))?;
-
-    get_sui_object_from_object_response::<SystemObject>(&response, system_object_id)
-        .map_err(|_| SuiClientError::WalrusSystemObjectDoesNotExist(system_object_id))?;
-
-    let object_pkg_id = get_package_id_from_object_response(&response)
-        .map_err(|_| SuiClientError::WalrusSystemObjectDoesNotExist(system_object_id))?;
+        .map_err(|error| {
+            tracing::debug!(%error, "unable to get the Walrus system object");
+            SuiClientError::WalrusSystemObjectDoesNotExist(system_object_id)
```
Comment (giac-mysten) [inline crates/walrus-sui/src/client/read_client.rs:302]: Remark: i guess the differentiation may be done with different error types. Calling everything `WalrusSystemObjectDoesNotExist` is a bit misleading. But good for now IMO.

File: Cargo.lock
Comment (mlegner) [reply Cargo.lock:1]: I believe that is due to a mistake in #731 where I ran the simtests before committing.

File: crates/walrus-sui/src/client/read_client.rs (hunk 1 above)
Comment (mlegner) [reply crates/walrus-sui/src/client/read_client.rs:302]: My idea there is to provide that extra info just in the debug messages as a normal user probably doesn't care too much about the differences.

**Pull request merged**